server/monitor.go

```diff
@@ -297,10 +303,17 @@ func (s *Server) Connz(opts *ConnzOptions) (*Connz, error) {
 		ci := &conns[i]
 		ci.fill(client, client.nc, c.Now)
 		// Fill in subscription data if requested.
-		if subs && len(client.subs) > 0 {
-			ci.Subs = make([]string, 0, len(client.subs))
-			for _, sub := range client.subs {
-				ci.Subs = append(ci.Subs, string(sub.subject))
+		if len(client.subs) > 0 {
+			if subsDet {
```

Comment:
I feel this "if/else" could be made a function of, because we use the same logic for routez..

Reply:
done